### PR TITLE
feat: React- HOC, DynamicTable Data and Add Trainee to store #41281

### PR DESCRIPTION
**Pull request by Priyankatiyar**
Created a new feature branch 41281.
[Refer Task](https://redmine.successive.tech/issues/41281)
listdata
TraineeAdded
SuccessfulLogin

### CODE REVIEW

File: training/src/pages/Trainee/TraineeList.jsx [hunk 1]
```diff
@@ -83,14 +88,47 @@ class TraineeList extends React.Component {
...
+    const value = this.context;
+    console.log('TraineeList value', value);
+    callApi({}, 'get', `trainee?skip=${skip}&limit=${limit}`).then((response) => {
+      console.log('List Response',response);
+      if (response.data === undefined) {
+        this.setState({
+          loading: false,
+          message: 'This is an error while displaying Trainee',
+        }, () => {
+          // const { message } = this.state;
```
Comment (shashank-nx) [inline training/src/pages/Trainee/TraineeList.jsx]: Remove the commented code.

File: training/src/pages/Trainee/components/AddDialog/AddDialog.jsx [hunk 2]
```diff
@@ -166,6 +210,6 @@ export default withStyles(stylePassword)(AddDialog);
 AddDialog.propTypes = {
   open: PropTypes.bool.isRequired,
   onClose: PropTypes.func.isRequired,
-  onSubmit: PropTypes.func.isRequired,
+  // onSubmit: PropTypes.func.isRequired,
```
Comment (shashank-nx) [inline training/src/pages/Trainee/components/AddDialog/AddDialog.jsx]: Remove the commented code.

**Review by shashank-nx — CHANGES_REQUESTED** [review]

File: training/src/pages/Trainee/TraineeList.jsx (hunk 1 above)
Comment (Priyankatiyar) [reply training/src/pages/Trainee/TraineeList.jsx]: Changes have been done .

**Review by shashank-nx — APPROVED** [review]
Looks fine to me

**Pull request merged**